In [67]:
#일원 선형회귀
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
df = pd.read_csv("regression.csv")



df

age  height  weight
0    1세    80.1    10.9
1    2세    87.8    13.2
2    3세    95.2    14.8
3    4세   102.3    16.7
4    5세   109.0    19.1
5    6세   115.5    21.5
6    7세   122.0    24.9
7    8세   127.8    28.5
8    9세   133.3    32.3
9   10세   138.0    35.4
10  11세   144.9    41.5
11  12세   150.7    45.2
12  13세   158.2    51.0
13  14세   164.7    57.1
14  15세   169.2    61.7
15  16세   170.3    63.4
16  17세   172.5    65.2
17  18세   172.9    67.4
18  19세   173.4    68.3

# fit(x,y) , predict(x) 
# x, y에 들어갈수 있는 타입 

##  1. numpy 2차원 배열
     df.height 는  series  ex) [80.1, 87.8, 95.2]
         1.1 np.array()를 이용 로 변경시 numpy 1차원 배열의 형태로 변경됨 ex)[80.1, 87.8, 95.2]
         1.2 reshape(-1,1) 사용 2차원 배열의 형태로 변경해야 함 :  [[80.1, 87.8, 95.2]]


In [68]:
#weight : 독립변수(x), height : 종속변수(y)
npLr=LinearRegression()
#1-1  
weightNpArray = np.array(df.weight)
heightNpArray = np.array(df.height)
#1-2
X_weightNpArrayD2 = weightNpArray.reshape(-1,1)
y_heightNpArrayD2 = heightNpArray.reshape(-1,1)
npModel = npLr.fit(X_weightNpArrayD2, y_heightNpArrayD2)    


In [69]:


X_testWeight=np.array([65]).reshape(-1,1)
y_testHeight=npModel.predict(X_testWeight)

print(npModel.coef_, npModel.intercept_)
y_testHeight

[[1.49421502]] [78.15367846]


array([[175.27765502]])

##    2. dataframe
    df.height 는  series ex) [80.1, 87.8, 95.2]
         2.1 데이터 프레임으로 변경
             df1 = pd.DataFrame(data=df.height.values, columns=['height'])
             df2 = pd.DataFrame({'height'df.height.values})
         

 

In [85]:
        
# wdf=pd.DataFrame(df.weight)
# hdf=pd.DataFrame(df.height)

#2-1
dfw = pd.DataFrame(data=df.weight.values, columns=['weight'])

dfh = pd.DataFrame({'height':df.height.values})

X_train, X_test, y_train, y_test = train_test_split(dfw, dfh, test_size=0.3, random_state=0)
dfLr=LinearRegression()
dfModel= dfLr.fit(X_train, y_train)    
len(y_test.index)

6

In [90]:
y_test=dfModel.predict(X_test)
print(X_test)


    weight
10    41.5
1     13.2
8     32.3
18    68.3
14    61.7
16    65.2


1.0

In [21]:
data=[1,2,3]
arr1=np.array(data)
arr1
arr1.shape
arr2=np.array([[1,2,3],[4,5,6],[7,8,9],[10,11,12]])
print(arr2)
arr2.shape

[[ 1  2  3]
 [ 4  5  6]
 [ 7  8  9]
 [10 11 12]]


(4, 3)